src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala

```
@@ -451,6 +451,84 @@ final case class TenantApiServiceImpl(
+  ): Route = authorize(ADMIN_ROLE) {
+    val operationLabel = s"Revoke certified attribute ${attributeId} to tenant $tenantId"
+    logger.info(operationLabel)
+
+    val now: OffsetDateTime = dateTimeSupplier.get()
+
+    val result: Future[Unit] = for {
+      requesterTenantUuid <- getOrganizationIdFutureUUID(contexts)
+      targetTenantUuid    <- tenantId.toFutureUUID
+      attributeUuid       <- attributeId.toFutureUUID
+      requesterTenant     <- tenantManagementService.getTenantById(requesterTenantUuid).map(_.toManagement)
+      certifierId         <- requesterTenant.features
+        .collectFirstSome(_.certifier.map(_.certifierId))
+        .toFuture(TenantIsNotACertifier(requesterTenantUuid))
+      attribute           <- attributeRegistryManagementService
+        .getAttributeById(attributeUuid)
+      _                   <- attribute.kind match {
+        case Certified => Future.unit
+        case _         => Future.failed(RegistryAttributeIdNotFound(attribute.id))
+      }
+      _                   <- attribute.origin match {
+        case Some(value) if (value == certifierId) => Future.unit
+        case _                                     =>
+          Future.failed(
+            CertifiedAttributeOriginIsNotComplaintToCertifier(
+              requesterTenantUuid,
+              targetTenantUuid,
+              attribute.origin,
+              certifierId
+            )
+          )
+      }
+      targetTenant        <- tenantManagementService.getTenantById(targetTenantUuid).map(_.toManagement)
+      attributeToRevoke   <- targetTenant.attributes
+        .mapFilter(_.certified)
+        .find(_.id == attributeUuid)
+        .toFuture(
+          CertifiedAttributeNotFoundInTenant(
+            targetTenant.id,
+            attribute.id,
+            attribute.origin.getOrElse("none"),
+            attribute.code.getOrElse("none")
+          )
+        )
+      revokedAttribute = attributeToRevoke.copy(revocationTimestamp = now.some)
```

@beetlecrunch do we need to return an error if the attribute is already revoked?
Or we treat it as an idempotent change?

---

Uhmmm, I prefer to return an error here because I don't want to overwrite the `revocationTimestamp`.

Do you have any concerns about it?

---

no, it makes sense 👍

---

If the behaviour needs to change, does has it to be propagate also for declared attribute or when revoked by M2M (m2mRevokeAttribute)?

---

we can leave the behaviour you implemented in this PR, no changes required